src/tests/dictionaryConcepts/actions/dictionaryConcept.test.js

```diff
@@ -1214,3 +1269,84 @@ describe('Add set mappings to concept', () => {
     });
   });
 });
+
+describe('buildNewMappingData', () => {
+  const fromConceptUrl = '/test/from/concept/url';
+  const toSourceUrl = '/test/to/source/url';
+  const toConceptCode = 'testCode';
+  const toConceptName = 'testName';
+  const mapType = MAP_TYPES_DEFAULTS[0];
+
+  it('should return the map_type, from_concept_url, to_source_url, to_concept_code, to_concept_name if given an external concept', () => {
+    const mapping = {
+      source: INTERNAL_MAPPING_DEFAULT_SOURCE,
+      map_type: mapType,
+      to_source_url: toSourceUrl,
+      to_concept_code: toConceptCode,
+      to_concept_name: toConceptName,
+    };
+    const expectedMapping = {
+      map_type: mapType,
+      from_concept_url: fromConceptUrl,
+      to_source_url: toSourceUrl,
+      to_concept_code: toConceptCode,
+      to_concept_name: toConceptName,
+    };
+    expect(buildNewMappingData(mapping, fromConceptUrl)).toEqual(expectedMapping);
+  });
+
+  it('should return the map_type, from_concept_url, to_concept_url, to_concept_name if given an internal concept', () => {
+    const mapping = {
+      source: 'not external',
+      map_type: mapType,
+      to_concept_code: toConceptCode,
+      to_concept_name: toConceptName,
+    };
+    const expectedMapping = {
+      map_type: mapType,
+      from_concept_url: fromConceptUrl,
+      to_concept_url: `${mapping.source}concepts/${toConceptCode}/`,
+      to_concept_name: toConceptName,
+    };
+    expect(buildNewMappingData(mapping, fromConceptUrl)).toEqual(expectedMapping);
+  });
+});
+
+describe('buildUpdateMappingData', () => {
+  const toSourceUrl = '/test/to/source/url';
+  const toConceptCode = 'testCode';
+  const toConceptName = 'testName';
+  const mapType = MAP_TYPES_DEFAULTS[0];
+
+  it('should return the map_type, to_source_url, to_concept_code, to_concept_name if given an external concept', () => {
+    const mapping = {
+      source: INTERNAL_MAPPING_DEFAULT_SOURCE,
+      map_type: mapType,
+      to_source_url: toSourceUrl,
+      to_concept_code: toConceptCode,
+      to_concept_name: toConceptName,
+    };
+    const expectedMapping = {
+      map_type: mapType,
+      to_source_url: toSourceUrl,
+      to_concept_code: toConceptCode,
+      to_concept_name: toConceptName,
+    };
+    expect(buildNewMappingData(mapping)).toEqual(expectedMapping);
+  });
+
+  it('should return the map_type, to_concept_url, to_concept_name if given an internal concept', () => {
```

Comment:
I have seen another "if given an internal concept" above. Why do these tests return different values?

Reply:
Because they are testing different methods.

Reply:
How do i tell that?

Reply:
They are in different describe blocks Daniel. Is this not descriptive enough?

Reply:
Have you thought of maintainers of this code would will be asking the same questions like me?

Reply:
I have updated these descriptions to include the names of the methods they are testing.

PS: I am not convinced that this makes it hard to understand the goal of the tests because, for example when jest displays failing tests, it displays the full context including under which describe blocks a test lies.
This in my opinion goes against the convention for structuring tests. As always, I'd love to be convinced otherwise. 🙂

Reply:
Would you rather that i first look at the body of a test to know what is exactly being tested? Or know it all by just looking at the description? 😊

Reply:
The latter, true.
But, I believe in this situation the location should be taken into context as well, that is which file and which describe block, don't you think?

Reply:
You have a good point in regards to that. So am convinced and merging. 😊